src/main/java/duke/functions/TaskList.java

```diff
@@ -0,0 +1,146 @@
+package duke.functions;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import duke.datatypes.Deadline;
+import duke.datatypes.Event;
+import duke.datatypes.Task;
+import duke.datatypes.Todo;
+
+public class TaskList {
+    private ArrayList<Task> arrayOfTasks = new ArrayList<>();
+    private final FileIO savedTextFile = new FileIO("data/savedtasklist.txt");
+    private int numberOfTasks;
+
+    public TaskList() {
+        setNumberOfTasks(0);
+    }
+
+    public int getNumberOfTasks() {
+        return numberOfTasks;
+    }
+
+    public void setNumberOfTasks(int numberOfTasks) {
+        this.numberOfTasks=numberOfTasks;
+    }
+
+    public void printAllTasks() {
+        if(arrayOfTasks.isEmpty()) {
+            System.out.println("OOPS!!! No task added yet.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i<arrayOfTasks.size();i++){
+                Task currentTask = arrayOfTasks.get(i);
+                System.out.println(i+1 + "." + currentTask.toString());
+            }
+        }
+    }
+
+    public void addTask(AnalyseCommand usercommands) {
+
+        Task newTask;
+
+        switch (usercommands.getCommandType()) {
+        case "todo":
+            newTask= new Todo(usercommands.getCommandDescription());
+            arrayOfTasks.add(newTask);
+            setNumberOfTasks(getNumberOfTasks() + 1);
+            printAddTaskMessage(newTask);
+            break;
+        case "deadline":
+            newTask= new Deadline(usercommands.getCommandDescription(), usercommands.getCommandTime());
+            arrayOfTasks.add(newTask);
+            setNumberOfTasks(getNumberOfTasks() + 1);
+            printAddTaskMessage(newTask);
+            break;
+        case "event":
+            newTask= new Event(usercommands.getCommandDescription(), usercommands.getCommandTime());
+            arrayOfTasks.add(newTask);
+            setNumberOfTasks(getNumberOfTasks() + 1);
+            printAddTaskMessage(newTask);
+            break;
+        case "error":
+            System.out.println("Please enter your command again!");
+            break;
+        default:
+            System.out.println("OOPS!! I'm sorry, but I don't know what that means :-(");
+            break;
+        }
+    }
+
+    private void printAddTaskMessage(Task newTask) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + newTask.toString());
+        if (getNumberOfTasks() < 2) {
+            System.out.println("Now you have " + getNumberOfTasks() + " task in the list.");
+        } else {
+            System.out.println("Now you have " + getNumberOfTasks() + " tasks in the list.");
```

Comment:
Both the bodies of the conditional have the same line of code. 2 Seems like a magic number here as well.